### ISSUE TITLE
Ionic serve getting hung up on webpack

### ISSUE THREAD

**Issue by brandyscarney** [driftyco/ionic-app-scripts#646]
_From @shaneparsons on January 12, 2017 20:38_

**Ionic version:**
[ ] **1.x**
[x] **2.x**

**I'm submitting a ...**
[x] bug report
[ ] feature request
[ ] support request

**Current behavior:**
When running ionic serve the process often hangs on `webpack started`, at which point the only thing I can do is quit / re-run ionic serve. This happens rather frequently.

**Expected behavior:**
The build process not getting hung up.

**Steps to reproduce:**
Run ionic serve -> make / save some changes to code (sometimes containing errors) -> repeat

**Other information:**
[Here's](https://forum.ionicframework.com/t/webpack-frequently-freezes-when-started/74226) a post on the forum describing the exact same issue I'm having.

**Ionic info:**
```
Cordova CLI: 6.4.0
Ionic Framework Version: 2.0.0-rc.1
Ionic CLI Version: 2.1.18
Ionic App Lib Version: 2.1.9
Ionic App Scripts Version: 0.0.43
ios-deploy version: 1.9.0
ios-sim version: 5.0.13
OS: macOS Sierra
Node Version: v7.2.1
Xcode version: Xcode 8.2.1 Build version 8C1002
```

_Copied from original issue: driftyco/ionic#10001_

**Comment by cozzbie** [comment]
Any luck?

**Comment by lleevvyy** [comment]
@danbucholtz I can confirm, that it happens from time to time. I will try to reproduce it, to give a hint  how to fix it, but it is for sure related with the fact, that one ts file is saved and just after that the same or another ts file is saved as well.

**Comment by shaneparsons** [comment]
Its still too early to say if this is a fix, but since rolling back to node 6.9.4 I've stopped having this issue. According to the readme, I don't believe node-sass supports node 7.*.

I'll update this if it happens again.

**Comment by danbucholtz** [comment]
I can honestly say I have never seen this with Node 6.x. I haven't tested Node 7.x yet, there are probably issues with dependencies that need to be updated. Please advise on how to recreate.

Thanks,
Dan

**Comment by shaneparsons** [comment]
@danbucholtz If my assumption is right, that it's due to the version of node-sass, since the latest RC still has the same version of node-sass, I imagine running `ionic serve` in any project w/ node 7+ installed will occasionally throw this behaviour.

**Comment by danbucholtz** [comment]
It very well could. For now, we don't recommend Node 7, we recommend Node 6 which is also what the Node federation recommends. Node 7 isn't stable yet and many deps don't work with it.

When I get a chance, I'll update our deps to hopefully get Node 7 working.

Thanks,
Dan

**Comment by cmaart** [comment]
Can still confirm. Hangs when doing changes while another change is still ongoing (being compiled).

```
Your system information:

 ordova CLI: 6.5.0
Ionic Framework Version: 2.0.1
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.1.0
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Windows 10
Node Version: v7.5.0
Xcode version: Not installed
```

**Comment by shaneparsons** [comment]
@cmaart downgrade to node 6.9.4

**Comment by corran2go** [comment]
also occurs on node 6.9.4 when saving multiple files while rebuild already in progress

`
Cordova CLI: 6.5.0 
Ionic Framework Version: 2.0.1
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.0.0
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Linux 4.4
Node Version: v6.9.4
Xcode version: Not installed
`

`
[14:08:28]  build started ... 
[14:08:28]  transpile update started ... 
[14:08:28]  transpile update finished in 58 ms 
[14:08:28]  webpack update started ... 
[14:08:34]  webpack update finished in 5.78 s 
[14:08:34]  build finished in 5.87 s 

[14:08:37]  build started ... 
[14:08:37]  transpile update started ... 
[14:08:37]  transpile update finished in 18 ms 
[14:08:37]  webpack update started ... 
[14:08:41]  webpack update finished in 4.40 s 
[14:08:41]  build finished in 4.42 s 

[14:08:45]  build started ... 
[14:08:45]  transpile started ... 
[14:08:48]  transpile finished in 3.35 s 
[14:08:48]  webpack update started ... 
[14:08:53]  build started ... 
[14:08:53]  template update started ... 
[14:08:53]  template update finished in 101 ms 
[14:08:53]  transpile update started ... 
[14:08:53]  transpile update finished in 15 ms 
[14:08:53]  webpack update started ... 
[14:08:58]  webpack update finished in 5.31 s 
[14:08:58]  build finished in 13.71 s 

[14:08:58]  build finished in 5.43 s 

[14:09:06]  build started ... 
[14:09:07]  transpile started ... 
[14:09:11]  transpile finished in 4.76 s 
[14:09:11]  webpack update started ... 
[14:09:21]  build started ... 
[14:09:21]  template update started ... 
[14:09:21]  template update finished in 145 ms 
[14:09:21]  transpile update started ... 
[14:09:21]  transpile update finished in 225 ms 
[14:09:21]  webpack update started ... 
[14:09:28]  webpack update finished in 6.39 s 
[14:09:28]  build finished in 21.17 s 

[14:09:28]  build finished in 6.77 s 
`

**Comment by mrbfrank** [comment]
I too experience frequent hanging on `ionic serve`

```
ionic info

Your system information:

Cordova CLI: 6.5.0
Ionic Framework Version: 2.0.1
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.0.0
ios-deploy version: 1.9.1
ios-sim version: 5.0.13
OS: macOS Sierra
Node Version: v7.5.0
Xcode version: Xcode 8.2.1 Build version 8C1002
```

**Comment by rlyttle** [comment]
I experience the same issue

Cordova CLI: 6.5.0
Ionic Framework Version: 2.1.0
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.0.0
ios-deploy version: Not installed
ios-sim version: Not installed
OS: OS X El Capitan
Node Version: v7.6.0
Xcode version: Xcode 8.2.1 Build version 8C1002

**Comment by shaneparsons** [comment]
@corran2go by "hung up" I meant stopping completely... your builds are finishing, so you aren't having the same issue.

@mrbfrank and @rlyttle downgrade to node 6.9.4.

**Comment by danbucholtz** [comment]
We should queue a build if there is an active build occurring. This should be doable. No timeline but we gotta fix this.

Thanks,
Dan

**Comment by danbucholtz** [comment]
This is fixed in `1.2.0`.

Thanks,
Dan

**Comment by glemiere** [comment]
I found this post randomly, did anybody tried to run `sudo xcodebuild -license` ? Because if I understand well, it does happen each time Apple is updating their "contract" (not native english speaker, actually mentally exhausted so not ready to google translate lol)
